scripts/release/generate_history_notes.py

```diff
@@ -100,8 +100,9 @@ def construct_core_history(component: str):
 
 
 def get_commits():
+    last_release = 'azure-cli-{}'.format(os.getenv('PRE_VERSION')) if os.getenv('PRE_VERSION') else 'upstream/release'
     out = subprocess.Popen([
-        'git', 'log', 'upstream/release...upstream/dev',
+        'git', 'log', '{}..upstream/dev'.format(last_release),
         '--pretty=format:"%H %s"'
     ],
```

Comment:
Compare dev commits with last release tag or release branch head, controlled by whether the env var `PRE_VERSION` is provided.